src/server.c

```diff
@@ -1930,6 +1939,11 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
         monotime el_duration = getMonotonicUs() - server.el_start;
         durationAddSample(EL_DURATION_TYPE_EL, el_duration);
         latencyTraceIfNeeded(server, eventloop, el_duration);
+
+        /* Accumulate time only for busy cycles */
+        if (!ProcessingEventsWhileBlocked && server.el_iteration_active) {
+            server.stat_active_time += el_duration;
+        }
```

Comment:
So that we don't count the expire cycle twice, we can move it to an else clause here.

```suggestion
        } else {
            server.stat_active_time += expire_cycle_time;
        }
```

To avoid complicated logic for ProcessingEventsWhileBlocked, we can simply skip the active expire cycle when ProcessingEventsWhileBlocked.

Reply:
Good catch, thanks!